Applications/DataExplorer/DataView/DiagramView/DiagramPrefsDialog.cpp

```diff
@@ -67,47 +67,52 @@ DiagramPrefsDialog::~DiagramPrefsDialog()
 
 void DiagramPrefsDialog::accept()
 {
-    if ((fromDateLine->text().length() > 0) && (toDateLine->text().length() > 0) &&
-        (!_list.empty()))
+    QDateTime start_date(DiagramList::getDateTime(fromDateLine->text()));
+    QDateTime end_date(DiagramList::getDateTime(toDateLine->text()));
+
+    if (start_date == QDateTime() || end_date == QDateTime() ||
+        start_date > end_date || _list.empty())
     {
-        // data has been loaded
-        if (_list[0]->size() > 0)
-        {
-            bool window_is_empty(false);
-            if (_window == nullptr)
-            {
-                _window = new DetailWindow();
-                _window->setAttribute(Qt::WA_DeleteOnClose);
-                window_is_empty = true;
-            }
+        OGSError::box("No data found...");
```

Comment:
It could be helpful for the user if the error message would be more specific. Error reasons could be
 - an invalid start or end date, 
 - an illegal date range,
 - an empty data set.
So I would split the if condition. In the error message I would also mention in which part of the  program the error occurred.

Reply:
As discussed, there is a lot that can go wrong if the format of the input file doesn't fit what I expect. I'd have to add lots of error handling and analysis and would rather not do it at this time because it'll take forever and this part of the Data Explorer is fairly unimportant anyway.